src/containers/ImportResources/ImportResources.js

```diff
@@ -57,32 +64,38 @@ export class ImportResources extends Component {
   };
 
   handleSubmit = event => {
-    const { namespace } = this.props;
+    const {
+      directory: applydirectory,
+      namespace,
+      repositoryURL: repourl,
+      serviceAccount: serviceaccount
+    } = this.state;
     const pipelinename = 'pipeline0';
     const gitresourcename = 'git-source';
     const gitcommit = 'master';
-    const applydirectory = this.state.directory;
-    const repourl = this.state.repositoryURL;
-    const serviceaccount = this.state.serviceAccount;
+
     const payload = {
-      pipelinename,
-      serviceaccount,
-      gitresourcename,
+      applydirectory,
       gitcommit,
+      gitresourcename,
+      pipelinename,
       repourl,
-      applydirectory
+      serviceaccount
     };
 
+    let hasError = false;
```

Comment:
get rid of has error and put 2 returns instead ?

Reply:
updated to use a single setState + return